posthog/test/test_instance_setting_model.py

```diff
@@ -67,3 +69,22 @@ def test_can_retrieve_multiple_settings(db):
         "MATERIALIZED_COLUMNS_ENABLED": True,
         "ASYNC_MIGRATIONS_AUTO_CONTINUE": True,
     }
+
+
+@pytest.mark.parametrize("cached_value", ("true", '["1:cool","abc:123"]', '"a_string"', "123"))
+def test_get_cached_instance_setting(db, cache, cached_value):
```

Comment:
Can we have a test for what happens when the postgres db call fails? There is a `simulate_postgres_error` or similar somewhere that you can use.

Reply:
Will do :+1:. Thanks for the comment, I wasn't aware of `simulate_postgres_error`.

Reply:
That one won't work for you since you're not using raw cursors, use 'greatest option 3' instead: https://neilkakkar.com/test-database-connection-django.html (or update it to do this!)

Reply:
I tried this using @neilkakkar's suggestion but modified the code a bit to only raise errors on queries to the `posthog_instancesettings` table*.

The results are that this catastrophically fails and we 500 with an unhandled exception. I assume that we don't want this. Any thoughts on what should be the solution on db being unavailable?:
1. Go to DLQ.
2. Default to environment variable (potentially miss any updates added dynamically).
3. Your suggestions?

*: The reason for this is that there is a query going before that to attempt to fetch the team id. On failure, the event is sent to the DLQ.

Reply:
Should default to the env var I think. It's an optimization from which we should fall back to the default behaviour I believe.